crates/driver/src/tests/cases/parallel_auctions.rs

```diff
@@ -13,28 +13,53 @@ use crate::tests::setup::{eth_order, eth_solution, setup, weth_pool};
 #[ignore]
 async fn driver_handles_solutions_based_on_id() {
     let order = eth_order();
-    let test = setup()
+    let mut test = setup()
         .pool(weth_pool())
         .order(order.clone())
         .solution(eth_solution())
+        .auction_id(1)
         .done()
         .await;
 
-    let id = test.solve().await.ok().id();
+    let solution_id = test.solve().await.ok().id();
 
     // calling `/reveal` or `/settle` with incorrect solution ids
     // results in an error.
     test.settle("99").await.err().kind("SolutionNotAvailable");
     test.reveal("99").await.err().kind("SolutionNotAvailable");
 
+    // calling `/reveal` or `/settle` with a reasonable id works
+    // but wrong auction id results in an error.
+    test.set_auction_id(100);
+    test.reveal(&solution_id)
+        .await
+        .err()
+        .kind("SolutionNotAvailable");
+    test.settle(&solution_id)
+        .await
+        .err()
+        .kind("SolutionNotAvailable");
+    test.set_auction_id(1);
+
     // calling `/reveal` or `/settle` with a reasonable id works.
```

Comment:
nit: adjusting the auction id logically belongs to the unit described by the comment
```suggestion

    // calling `/reveal` or `/settle` with a reasonable id works.
    test.set_auction_id(1);
```